pkg/tinydtls/contrib/sock_dtls.c

```diff
@@ -642,8 +642,17 @@ ssize_t sock_dtls_send_aux(sock_dtls_t *sock, sock_dtls_session_t *remote,
         }
     }
 
-    res = dtls_write(sock->dtls_ctx, &remote->dtls_session,
-                     (uint8_t *)data, len);
+    res = 0;
+    while (snips) {
+        int tmp = dtls_write(sock->dtls_ctx, &remote->dtls_session,
+                             snips->iol_base, snips->iol_len);
```

Comment:
Does this actually work? IIRC there was something about continuous buffers with DTLS.

Reply:
Oops, you are of course right!
I only tested the single-snip case as I didn't have the multi-snip use case (for zero-copy CoAP over DTLS payload) ready yet.

Now this requires a patch to tinydtls - fortunately it already can handle non-continuous buffers internally: [#159](https://github.com/eclipse/tinydtls/pull/159)